vsintegration/src/FSharp.Editor/Navigation/GoToDefinitionService.fs

```diff
@@ -30,83 +34,292 @@ type internal FSharpNavigableItem(document: Document, textSpan: TextSpan) =
         member this.DisplayTaggedParts = ImmutableArray<TaggedText>.Empty
         member this.ChildItems = ImmutableArray<INavigableItem>.Empty
 
-[<Shared>]
-[<ExportLanguageService(typeof<IGoToDefinitionService>, FSharpCommonConstants.FSharpLanguageName)>]
-type internal FSharpGoToDefinitionService 
-    [<ImportingConstructor>]
-    (
-        checkerProvider: FSharpCheckerProvider,
-        projectInfoManager: ProjectInfoManager,
-        [<ImportMany>]presenters: IEnumerable<INavigableItemsPresenter>
-    ) =
-
-    static member FindDefinition(checker: FSharpChecker, documentKey: DocumentId, sourceText: SourceText, filePath: string, position: int, defines: string list, options: FSharpProjectOptions, textVersionHash: int) : Async<Option<range>> = 
-        asyncMaybe {
-            let textLine = sourceText.Lines.GetLineFromPosition(position)
-            let textLinePos = sourceText.Lines.GetLinePosition(position)
+
+module internal FSharpGoToDefinition =
+  
+    /// Parse and check the provided document and try to find the defition of the symbol at the position
+    let checkAndFindDefinition
+        (checker: FSharpChecker, documentKey: DocumentId, sourceText: SourceText, filePath: string, position: int,
+         defines: string list, options: FSharpProjectOptions, textVersionHash: int) : Option<range> = maybe {
```

Comment:
I understand that we pass so many stuff here in order to make it tastable, because we cannot use `Document` here. But this bunch of parameters are passed to many functions in FSharp.Editor project. What about a new data type, like this:
```fsharp
type EditorDocument =
  { FilePath: string
    Key: DocumentId
    Text: SourceText
    TextVersionHash: int 
    Defines: string[] }
```
?